features/recipes/operations.py

```diff
@@ -240,6 +239,22 @@ def update_recipe(recipe_id: int) -> None:
     logging.info(f"Recipe #{recipe_id} was updated")
 
 
+def patch_recipe(recipe_id: int, field: str, value: str, updated_by: str = 'me') -> Type[Recipe]:
```

Comment:
I will raise recipe not found if the recipe is not published or if the requestor is not the author or admin. 
Add authentication to patch route and pass the user hire, than pass it to get_recipe_by_id